applications/potree-viewer/tapisjob_app.sh

```diff
@@ -0,0 +1,59 @@
+handle_error() {
+  local EXITCODE=$1
+  echo "Potree Viewer job exited with an error status. $EXITCODE" >&2
+  echo "$EXITCODE" > "${_tapisExecSystemOutputDir}/tapisjob.exitcode"
+  exit $EXITCODE
+}
+
+set -x
+echo "TACC: job $_tapisJobUUID execution at: `date`"
+
+LOGIN_PORT=$(( ((RANDOM<<15)|RANDOM) % 100 + 5900 ))
+quit=0
+
+while [ "$quit" -ne 1 ]; do
+  netstat -a | grep $LOGIN_PORT >> /dev/null
+  if [ $? -gt 0 ]; then
+    quit=1
+  else
+    LOGIN_PORT=`expr $LOGIN_PORT + 1`
+  fi
+done
+echo "Using port=$LOGIN_PORT"
+
+# Webhook callback url for job ready notification
+# (notifications sent to INTERACTIVE_WEBHOOK_URL (i.e. https://3dem.org/webhooks/interactive/))`
+INTERACTIVE_WEBHOOK_URL="${_INTERACTIVE_WEBHOOK_URL}"
+
+#connect to interactive session on VM
+curl -k --data "event_type=interactive_session_ready&address=https://wma-exec-01.tacc.utexas.edu:${LOGIN_PORT}&owner=${_tapisJobOwner}&job_uuid=${_tapisJobUUID}" "${_INTERACTIVE_WEBHOOK_URL}" &
+
+echo "TACC: Your interactive session is now running!"
+echo "TACC: Connect to your session at: https://wma-exec-01.tacc.utexas.edu:${LOGIN_PORT}/"
+
+
+apptainer run \
```

Comment:
The reason for archiving issue is - apptainer is running in foreground. 

Tapis timed out. See this note from tapis logs:
```
 edu.utexas.tacc.tapis.jobs.exceptions.JobException: JOBS_MONITOR_EARLY_TERMINATION ZipNativeMonitor terminated monitoring early for job 0d7c9280-071c-4aa6-9fc4-a5a89c09fc94-007 for policy reason TIME_EXPIRED, job outcome is FAILED_SKIP_ARCHIVE.
```

Can you run this in background.

Reply:
Also, because of that even with tapis end the job on their end, wma-exec-01 is still running apptainer.

Reply:
Thank you! Apptainer now runs in the background, and the maxMinutes is enforced and archiving is functioning as expected.